### PR TITLE
[FIX] Resource: Keep stats size up to date

### PR DESCRIPTION
**Pull request by tobiasso85**
Ensure statInfo.size is always set when resource's content is modified.

### CODE REVIEW

File: test/lib/Resource.js [hunk 1]
```diff
@@ -142,6 +142,58 @@ test("Resource: setString", (t) => {
...
+
+	const resource = new Resource({
+		path: "my/path/to/resource"
+	});
+
+	resource._setSize(1337);
+
+	t.is(resource.getStatInfo().size, 1337);
+});
+
```
Comment (matz3) [inline test/lib/Resource.js:196]: We should also add test cases when passing a real `statInfo` to the constructor (from fs.stat) and then changing the content. This is the use case when a resource is created by the FsAdapter.

Comment (tobiasso85) [reply test/lib/Resource.js:198]: done

File: lib/Resource.js [hunk 2]
```diff
@@ -76,6 +76,9 @@ class Resource {
 		if (typeof string === "string" || string instanceof String) {
```
[the remark this answers is not in the document]
Comment (tobiasso85) [reply lib/Resource.js:76]: done

**Comment by matz3** [comment]
Maybe this is even a feature, to update the size when updating the content?

**Comment by RandomByte** [comment]
As discussed today with @matz3 and @codeworrior: This does not work for resources where the content is updated using `setStream()`. For example [processors/stringReplacer.js](https://github.com/SAP/ui5-builder/blob/master/lib/processors/stringReplacer.js) modifies the content in a streaming way.

A possible alternative solution could be to provide an asynchronous `getSize()` API which can transform a stream into a buffer and calculate the size if required.

**Pull request merged**